File: 06/problem.py
def getInput(file):
    data = None

    with open(file) as f:
        data = f.read().strip()

    return data

def solveFirst(data):
    n = len(data)
    for idx in range(4, n):
        chars = data[idx-4:idx]
        if (len(set(chars))== 4):
            print(chars, idx)
            return

def solveSecond(data):
    n = len(data)
    for idx in range(14, n):
        chars = data[idx-14:idx]
        if (len(set(chars))== 14):
            print(chars, idx)
            return

if __name__ == "__main__":
    data = getInput("./input")
    solveFirst(data)
    solveSecond(data)

File: 15/problem.lua
local function split (inputstr, sep)
    if sep == nil then
       sep = "%s"
    end
    local t={}
    for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
       table.insert(t, str)
    end
    return t
 end

local function getInput(fileName)
    local fileIn = io.open(fileName)

    -- check if file was read successfully
    if (fileIn == nil)
    then
        return
    end

    -- this has state, so can't use twice for each problem 
    local lines = fileIn:lines()
    local count = 1
    local returnLines = {}

    -- i want a stateless iterator, theres probably a better way to do this but it works
    for line in lines do
        returnLines[count] = line
        count = count + 1
    end

    return returnLines
end

-- i have to write a set function??? lul
function Set(str)
    local s = {}
    for char in str:gmatch"." do s[char] = true end
    return s
end

function IsUpper(char)
    return string.match(char, '%u')
end

local function solveFirst(input)
    -- do things here to solve problem
    for idx, line in pairs(input) do
        print(line)
    end
    -- do things above here to solve the problem
end

local function solveSecond(input)
    -- do things here to solve problem
    for idx, line in pairs(input) do
        print(line)
    end
    -- do things above here to solve the problem
end

local input = getInput("./test-input")
solveFirst(input)
solveSecond(input)
